In [8]:
import pandas as pd 
from scipy import stats
from scipy.stats import chi2_contingency,pearsonr
from sklearn.preprocessing import LabelEncoder

In [9]:
data=pd.read_excel("Adidas US Sales Datasets.xlsx")

In [10]:
data. head()

Retailer  Retailer ID Invoice Date     Region     State      City  \
0  Foot Locker      1185732   2020-01-01  Northeast  New York  New York   
1  Foot Locker      1185732   2020-01-02  Northeast  New York  New York   
2  Foot Locker      1185732   2020-01-03  Northeast  New York  New York   
3  Foot Locker      1185732   2020-01-04  Northeast  New York  New York   
4  Foot Locker      1185732   2020-01-05  Northeast  New York  New York   

                     Product  Price per Unit  Units Sold  Total Sales  \
0      Men's Street Footwear            50.0        1200     600000.0   
1    Men's Athletic Footwear            50.0        1000     500000.0   
2    Women's Street Footwear            40.0        1000     400000.0   
3  Women's Athletic Footwear            45.0         850     382500.0   
4              Men's Apparel            60.0         900     540000.0   

   Operating Profit  Operating Margin Sales Method  
0          300000.0              0.50     In-store  
1          150000.0              0.30     In-store  
2          140000.0              0.35     In-store  
3          133875.0              0.35     In-store  
4          162000.0              0.30     In-store

In [11]:
contingency_table=pd.crosstab(data['Region'], data['Sales Method'])
contingency_table

Sales Method  In-store  Online  Outlet
Region                                
Midwest            445    1165     262
Northeast          683     936     757
South               20     844     864
Southeast          194     792     238
West               398    1152     898

In [12]:
chi2, pval, dof, expected = chi2_contingency(contingency_table)

In [13]:
print("Chi-square statistic:", chi2)
print("p-value:", pval)
print("Degrees of freedom:", dof)

if pval < 0.05:
    print("There is a relationship between Region and Sales Method (p =", pval, ").")
else:
    print("There is no relationship between Region and Sales Method (p =", pval, ").")


Chi-square statistic: 1079.9201605996154
p-value: 8.311234551317755e-228
Degrees of freedom: 8
There is a relationship between Region and Sales Method (p = 8.311234551317755e-228 ).


The chi-square statistic is quite high, indicating a strong association between the Region and Sales store.

In [14]:

data['Region']=LabelEncoder().fit_transform(data['Region'])
data['Sales Method']=LabelEncoder().fit_transform(data['Sales Method'])
correlation_coefficient, p_value = pearsonr(data['Region'], data['Sales Method'])
print("Pearson correlation coefficient:", correlation_coefficient)
print("P-value:", p_value)

Pearson correlation coefficient: 0.13647748175617958
P-value: 2.455603830038244e-41


The correlation coefficient of approximately 0.1365 suggests a weak positive correlation.The low p-value indicates that the observed correlation is unlikely to be due to random chance.